pkg/services/store/object/object.proto

```diff
@@ -250,16 +241,43 @@ message ObjectSearchRequest {
   // Sorting instructions `field ASC/DESC`
   repeated string sort = 7;
 
-  // TODO, limit the set of fields we actually want returned
-  // Only supported in the QueryResponse flavor?
-  repeated string fields = 8;
+  // Return the full body in each payload
+  bool with_body = 8;
 
   // Return the full body in each payload
-  bool with_body = 9;
+  bool with_summary = 9;
+}
+
+// Search result metadata for each object
+message ObjectSearchResult {
+  // Unique ID
+  string UID = 1;
+
+  // Identify the object kind.  This kind will be used to apply a schema to the body and
+  // will trigger additional indexing behavior.
+  string kind = 2;
+
+  // The current veresion of this object
+  string version = 3;
+
+  // Display name from summary
+  string name = 4;
+
+  // Time in epoch milliseconds that the object was updated
```

Comment:
NOTE: the query results do not include the the 'created' time -- Most of the time, I think that is fine, we can and should include a more flexible response that allows picking explicit fields (and setting the results in a DataFrame)